lib/crypt.c

```diff
@@ -320,7 +321,12 @@ crypt_checksalt (const char *setting)
   const struct hashfn *h = get_hashfn (setting);
 
   if (h)
-    retval = CRYPT_SALT_OK;
+    {
+      retval = CRYPT_SALT_OK;
+
+      if (h->is_strong == 0)
+        retval = CRYPT_SALT_METHOD_LEGACY;
+    }
```

Comment:
I would probably have written this as

```
if (h)
  retval = h->is_strong ? CRYPT_SALT_OK : CRYPT_SALT_METHOD_LEGACY;
```

but I don't insist on that change.

Reply:
Yeah, I didn't use the ternary operator here on purpose, as we would likely need to change to the style I used as soon as we start implementing one of the other return codes.